spark3-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ExtendedDataSourceV2Strategy.scala

```diff
@@ -75,6 +76,8 @@ case class ExtendedDataSourceV2Strategy(spark: SparkSession) extends Strategy {
     case ReplaceData(_, batchWrite, query) =>
       ReplaceDataExec(batchWrite, planLater(query)) :: Nil
 
+    case MergeInto(mergeIntoProcessor, targetRelation, child) =>
+      MergeIntoExec(mergeIntoProcessor, targetRelation, planLater(child)) :: Nil
```

Comment:
Nit: to make this class more readable, we've been separating cases with a blank line.